bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -571,45 +568,37 @@ private void publishSuspectedLedgers(String bookieIP, Set<Long> ledgers)
         }
 
         public void operationComplete(int rc, Set<LedgerFragment> fragments) {
-            try {
-                if (rc == BKException.Code.OK) {
-                    Set<BookieSocketAddress> bookies = Sets.newHashSet();
-                    for (LedgerFragment f : fragments) {
-                        bookies.addAll(f.getAddresses());
-                    }
-                    for (BookieSocketAddress bookie : bookies) {
-                        publishSuspectedLedgers(bookie.toString(), Sets.newHashSet(lh.getId()));
-                    }
-                }
-                lh.close();
-            } catch (BKException bke) {
-                LOG.error("Error closing lh", bke);
-                if (rc == BKException.Code.OK) {
-                    rc = BKException.Code.ReplicationException;
-                }
-            } catch (InterruptedException ie) {
-                LOG.error("Interrupted publishing suspected ledger", ie);
-                Thread.currentThread().interrupt();
-                if (rc == BKException.Code.OK) {
-                    rc = BKException.Code.InterruptedException;
-                }
-            } catch (BKAuditException bkae) {
-                LOG.error("Auditor exception publishing suspected ledger", bkae);
-                if (rc == BKException.Code.OK) {
-                    rc = BKException.Code.ReplicationException;
+            if (rc == BKException.Code.OK) {
+                Set<BookieSocketAddress> bookies = Sets.newHashSet();
+                for (LedgerFragment f : fragments) {
+                    bookies.addAll(f.getAddresses());
                 }
+                publishSuspectedLedgersAsync(
+                    bookies.stream().map(BookieSocketAddress::toString).collect(Collectors.toList()),
+                    Sets.newHashSet(lh.getId())
+                ).whenComplete((result, cause) -> {
+                    if (null != cause) {
+                        callback.processResult(Code.ReplicationException, null, null);
```

Comment:
just want to make sure that this 'cause' is not lost in log errors. will it not hide cause completely here?